src/control/server/storage/scm/provider.go

```diff
@@ -562,9 +565,18 @@ func (p *Provider) CheckFormat(req FormatRequest) (*FormatResponse, error) {
 		}
 
 		p.log.Debugf("device %s filesystem: %s", req.Dcpm.Device, fsType)
-		if fsType != fsTypeNone {
-			return res, nil
+
+		switch fsType {
+		case fsTypeExt4:
```

Comment:
What happens if you get a FS type thats not none or ext4? Can you format these appdirect namespaces as something other than ext4 with dax?

Reply:
In that case we return a result with Formatted: true/Mountable: false, and it's up to the caller to decide what to do with it. Up in IOServerInstance.NeedsScmFormat(), we interpret this result as meaning that SCM needs to be reformatted by the control plane in order to be usable. So the admin would issue `dmg storage format --reformat` in order to clear the previous filesystem.